content/guides/typescript-for-solid.mdx

```diff
@@ -578,12 +579,65 @@ let [name, setName] = createSignal("");
 <input type="text" use:model={[name, setName]} />;
 ```
 
+In using `DirectiveFunctions`, there's the ability to check both arguments (if present) by detailing the entire function type:
+
+```tsx
+function model(element: HTMLInputElement, value: Accessor<Signal<string>>) {
+  const [field, setField] = value();
+  createRenderEffect(() => (element.value = field()));
+  element.addEventListener("input", (e) => setField(e.target.value));
+}
+
+function log(element: DOMElement) {
+  console.log(element);
+}
+
+let num = 0;
+function count() {
+  num++;
+}
+
+function foo(comp: number, ...args: string[]) {
```

Comment:
This actually shouldn't work (`comp` can't be a `number`), but it does because of a bug.